device/common/include/traccc/device/container_d2h_copy_alg.hpp

```diff
@@ -0,0 +1,67 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/utils/algorithm.hpp"
+#include "traccc/utils/memory_resource.hpp"
+
+// VecMem include(s).
+#include <vecmem/utils/copy.hpp>
+
+namespace traccc::device {
+
+/// Algorithm to copy a container from a device to the host
+///
+/// Specialisations of this algorithm can be used to get a container back
+/// from a device to the host. Ideally at the end of an algorithm chain that
+/// would've produced something in device memory.
+///
+/// Note that the algorithm needs to treat constant views and buffers a little
+/// differently, so the user should make sure to pass in the correct type.
+///
+/// @tparam CONTAINER_TYPES One of the "container types" traits
+///
+template <typename CONTAINER_TYPES>
+class container_d2h_copy_alg
+    : public algorithm<typename CONTAINER_TYPES::host(
+          const typename CONTAINER_TYPES::const_view&)>,
+      public algorithm<typename CONTAINER_TYPES::host(
+          const typename CONTAINER_TYPES::buffer&)> {
```

Comment:
Is it too much to ask to have support for passing in data by rvalue reference? :angel:

Reply:
:confused: Instead of just arguing with you, here you go. Now you explain to me how this version of the code is better... As I truly believe it's worse like this. (Harder to maintain and more error prone for no performance gain.)